deployer/hub.py

```diff
@@ -522,54 +527,57 @@ def deploy(self, auth_provider, secret_key, skip_hub_health_test=False):
         """
         Deploy this hub
         """
-        # Find helm chart values files
-        cluster_dir = Path(os.getcwd()).joinpath(
-            "config", "clusters", self.cluster.spec["name"]
-        )
-        values_files = [
-            f"--values={cluster_dir.joinpath(values_file)}"
-            for values_file in self.spec["helm_chart_values_files"]
-        ]
-
-        # Ensure helm charts are up to date
-        os.chdir("helm-charts")
-        subprocess.check_call(["helm", "dep", "up", "basehub"])
-        if self.spec["helm_chart"] == "daskhub":
-            subprocess.check_call(["helm", "dep", "up", "daskhub"])
-        os.chdir("..")
-
-        # Check if this cluster has any secret config. If yes, read it in.
-        secret_config_path = Path(os.getcwd()).joinpath(
-            "secrets", "config", "clusters", f'{self.cluster.spec["name"]}.cluster.yaml'
-        )
+        # Support overriding domains in case we want to block access to a hub
+        #
+        # Check if this hub has an override file. If yes, apply overrides.
```

Comment:
I think this configuration is too specific and we should have a more generalized strategy to support secret cluster.yaml configuration. But, I also think we should iterate on that later and get this PR merged sooner rather than later. Due to that, I made the comment more explicit to help me understand it better and added a fixme that represents what I think we should do in the long term.

```suggestion
        # Support overriding domain configuration in the loaded cluster.yaml via
        # a cluster.yaml specified enc-<something>.secret.yaml file that only
        # includes the domain configuration of a typical cluster.yaml file.
        #
        # Check if this hub has an override file. If yes, apply override.
        #
        # FIXME: This could could be generalized so that the cluster.yaml would allow
        #        any of this configuration to be specified in a secret file instead of a
        #        publicly readable file. We should not keep adding specific config overrides
        #        if such need occur but instead make cluster.yaml be able to link to
        #        additional secret configuration.
```

Reply:
Yeah, I went back and forth on this change and whether to make it general or not. On the one hand, we only do this for one hub, on the other hand, now we have an explicit list of helm chart values files (including secret ones), there's not much left at the cluster.yaml level we'd want to explicitly override.

- For `auth0`, we override by setting `auth0.enabled = false` and provide alternative auth config using a JupyterHub native OAuthenticator in the helm chart values, which would be explicitly listed
- For the hub `name` (helm namespace and release name), I'm not sure why we would ever want to override this?
- `helm_chart` we can't override without deleting a release and reinstalling

So I agree it's specific, but I don't know if we'd get much more benefit from a completely generalised version of this?